frontend/src/common/PageWithTabs.tsx

```diff
@@ -36,9 +39,15 @@ export default function PageWithTabs({
 
   const handleChange = (_event: SyntheticEvent, newValue: string) => {
     setCurrentTab(newValue)
-    router.replace({
-      query: { ...router.query, tab: newValue },
-    })
+    if (requiredUrlParams) {
```

Comment:
If you set a default value for `requiredUrlParams` of `{}` then you can remove the 'if' statement and simplify the code.